src/libsystemd/sd-id128/sd-id128.c

```diff
@@ -181,3 +182,34 @@ _public_ int sd_id128_randomize(sd_id128_t *ret) {
         *ret = make_v4_uuid(t);
         return 0;
 }
+
+_public_ int sd_id128_get_machine_app_specific(sd_id128_t app_id, sd_id128_t *ret) {
+        _cleanup_(sd_hash_unrefp) sd_hash *h = NULL;
+        sd_id128_t m, result;
+        const void *p;
+        int r;
+
+        assert_return(ret, -EINVAL);
+
+        r = sd_id128_get_machine(&m);
+        if (r < 0)
+                return r;
+
+        r = sd_hash_new_with_key(&h, "hmac(sha256)", &m, sizeof(m));
+        if (r < 0)
+                return r;
+
+        r = sd_hash_put(h, &app_id, sizeof(app_id));
+        if (r < 0)
+                return r;
+
+        r = sd_hash_digest_data(h, &p);
+        if (r < 0)
+                return r;
+
+        /* We chop off the trailing 16 bytes */
+        memcpy(&result, p, MIN(sd_hash_get_size(h), sizeof(result)));
```

Comment:
Why exactly are there 16 extra bytes at the end of `digest_data`? What does it represent (or, alternately, why isn't it truncated by `sd_hash_digest_data()` rather than by the consumer?